src/ansys/grantami/bomanalytics/_item_results.py

```diff
@@ -56,7 +58,7 @@ def create_material_impacted_substances_result(
         item_result = MaterialWithImpactedSubstancesResult(
             reference_type=reference_type,
             reference_value=result_with_impacted_substances.reference_value,
-            legislations=result_with_impacted_substances.legislations,
+            legislations=_cast_unset_union_to_any(result_with_impacted_substances.legislations),
```

Comment:
Lots of changes here, mainly related to either converting things that are optional to things that we want to be nullable in the idiomatic layer, or just stripping the optional 'Unset_Type' type from things that we know are actually required, but just not specified as such in the API

Reply:
My chosen approach was to do this as high up in the call stack as possible, to try and avoid changing signatures of functions that we know are good.

Reply:
Is `_cast_*` the right terminology here?
I was initially confused as to what it would do, because I thought it worked like the builtin typing.cast

Reply:
I suppose I've used 'cast' in the sense that it will convert the type of something that contains a 'unset' type to something that does not contain an 'unset' type. Typically if a type conversion cannot complete for some reason, then it will raise an exception.

In this case, it will raise an exception because it's not possible to cast the value `Unset` to something that's not of `Unset_Type`.